### DISCUSSION TITLE
React development tools - can't select components in iframe

### DISCUSSION THREAD

**Discussion by Lovor01** [WordPress/gutenberg#67700] [Q&A]
Since a while, I cant get react developer tools to work properly. I can't select components which are inside editor iframe. It does not work on Firefox, nor in Chrome. This is a major nuissance to me. Any suggestions how to fix this?
(Windows machine, latest browser versions and developer tools, as well as latest @wordpress/scripts)

**Comment by stokesman** [comment]
Yes this can be quite a bother. One thing I’ve noticed is that using the browser dev tool built-in element picker causes React dev tools to also pick the associated component. You have to flip back to the components tab but in my experience the correct component is picked. In Chrome it can be a bit annoying because the components list can appear blank (I think the scroll position is in limbo or idk) but I found clicking the blank space gets it in sync so you can inspect surrounding components.

**Comment by Lovor01** [comment]
@stokesman 's solution works sometimes (thanks for sharing that!), sometimes I have to repeat that several times to work - it is a nuissance, but I can live with it like that.

**Comment by stokesman** [comment]
By the way, I’d already looked into whether React dev tools can support this and it sounds like it can https://github.com/facebook/react-devtools/issues/76#issuecomment-128091900. Yet the iframe’s document has to load with a script that executes before react-dom itself. The editor’s iframe is a dynamic document and so I think there’s no way to make it work here.
